osmosis-installer.py

```diff
@@ -161,6 +189,142 @@ def cosmovisorInit ():
+    if startNow == "1":
+        subprocess.run(["clear"], shell=True)
+        cosmovisorService()
+        subprocess.run(["sudo systemctl start cosmovisor"], shell=True, env=my_env)
+        replayComplete()
+    if startNow == "2":
+        subprocess.run(["clear"], shell=True)
+        replayDelay()
+    else:
+        subprocess.run(["clear"], shell=True)
+        startReplayNow()
+
+
+def replayFromGenesisLevelDb ():
+    print(bcolors.OKGREEN + "Setting Up Cosmovisor..." + bcolors.ENDC)
+    os.chdir(os.path.expanduser(HOME))
+    subprocess.run(["go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@v1.0.0"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, env=my_env)
+    subprocess.run(["mkdir -p "+osmo_home+"/cosmovisor"], shell=True, env=my_env)
+    subprocess.run(["mkdir -p "+osmo_home+"/cosmovisor/genesis"], shell=True, env=my_env)
+    subprocess.run(["mkdir -p "+osmo_home+"/cosmovisor/genesis/bin"], shell=True, env=my_env)
+    subprocess.run(["mkdir -p "+osmo_home+"/cosmovisor/upgrades"], shell=True, env=my_env)
+    subprocess.run(["mkdir -p "+osmo_home+"/cosmovisor/upgrades/v4/bin"], shell=True, env=my_env)
+    subprocess.run(["mkdir -p "+osmo_home+"/cosmovisor/upgrades/v5/bin"], shell=True, env=my_env)
+    subprocess.run(["mkdir -p "+osmo_home+"/cosmovisor/upgrades/v7/bin"], shell=True, env=my_env)
+    os.chdir(os.path.expanduser(HOME+'/osmosis'))
+    print(bcolors.OKGREEN + "Preparing v4 Upgrade..." + bcolors.ENDC)
+    subprocess.run(["git checkout v4.2.0"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, env=my_env)
+    subprocess.run(["make build"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, env=my_env)
+    subprocess.run(["cp build/osmosisd "+osmo_home+"/cosmovisor/upgrades/v4/bin"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, env=my_env)
+    print(bcolors.OKGREEN + "Preparing v5/v6 Upgrade..." + bcolors.ENDC)
+    subprocess.run(["git checkout v6.4.1"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, env=my_env)
+    subprocess.run(["make build"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, env=my_env)
+    subprocess.run(["cp build/osmosisd "+osmo_home+"/cosmovisor/upgrades/v5/bin"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, env=my_env)
+    print(bcolors.OKGREEN + "Preparing v7 Upgrade..." + bcolors.ENDC)
+    subprocess.run(["git checkout v7.0.3"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, env=my_env)
+    subprocess.run(["make build"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, env=my_env)
+    subprocess.run(["cp build/osmosisd "+osmo_home+"/cosmovisor/upgrades/v7/bin"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, env=my_env)
+    subprocess.run(["git checkout v3.1.0"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, env=my_env)

```

Comment:
Missing: `print(bcolors.OKGREEN + "Preparing v3.1.0 Upgrade..." + bcolors.ENDC)`

Reply:
Nvm, I see why we do it this way